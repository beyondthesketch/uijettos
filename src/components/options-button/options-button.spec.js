import React from 'react';
import renderer from 'react-test-renderer';

import UijettosOptionsButton from './options-button';

describe('UijettosOptionsButton component', () => {
    test('Renders correctly when not supplied any props', () => {
        const tree =
            renderer
                .create(
                    <UijettosOptionsButton />
                )
                .toJSON();

        expect(tree).toMatchSnapshot();
    });

    test('Renders correctly when supplied a css prefix', () => {
        const tree =
            renderer
                .create(
                    <UijettosOptionsButton
                        cssClassPrefix="doodads"
                        options={
                            new Map(
                                [
                                    ['widgets', () => null],
                                    ['doodads', () => null]
                                ]
                            )
                        }
                    />
                );

        expect(tree.toJSON()).toMatchSnapshot();

        tree.root.instance.setState(
            {
                open: true
            }
        );

        expect(tree.toJSON()).toMatchSnapshot();
    });

    test('Renders correctly when supplied with a label', () => {
        const tree =
            renderer
                .create(
                    <UijettosOptionsButton
                        label={ 'Foo Bar' }
                    />
                )
                .toJSON();

        expect(tree).toMatchSnapshot();
    });

    test('Renders correctly when supplied with options', () => {
        const tree =
            renderer
                .create(
                    <UijettosOptionsButton
                        options={
                            new Map(
                                [
                                    ['widgets', () => null],
                                    ['doodads', () => null]
                                ]
                            )
                        }
                    />
                )

        expect(tree.toJSON()).toMatchSnapshot();

        tree.root.instance.setState(
            {
                open: true
            }
        );

        expect(tree.toJSON()).toMatchSnapshot();
    });

    test('Renders options correctly when trigger button is clicked', () => {
        const mockEvent = {
            stopPropagation: jest.fn()
        };
        const tree =
            renderer
                .create(
                    <UijettosOptionsButton
                        options={
                            new Map(
                                [
                                    ['widgets', () => null],
                                    ['doodads', () => null]
                                ]
                            )
                        }
                    />
                );

        tree.root
            .findByProps(
                {
                    className: 'uijettos-options-button__options-trigger'
                }
            )
            .props
            .onClick(
                mockEvent
            );

        expect(tree.toJSON()).toMatchSnapshot();
    });
});
