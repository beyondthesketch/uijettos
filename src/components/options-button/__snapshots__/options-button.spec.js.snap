// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UijettosOptionsButton component Renders correctly when not supplied any props 1`] = `
<div
  className="uijettos-options-button"
  onBlur={[Function]}
  onFocus={[Function]}
>
  <button
    className="uijettos-options-button__main-button"
  />
  <button
    className="uijettos-options-button__options-trigger"
    onClick={[Function]}
  />
</div>
`;

exports[`UijettosOptionsButton component Renders correctly when supplied a css prefix 1`] = `
<div
  className="doodads-options-button"
  onBlur={[Function]}
  onFocus={[Function]}
>
  <button
    className="doodads-options-button__main-button"
  />
  <button
    className="doodads-options-button__options-trigger"
    onClick={[Function]}
  />
</div>
`;

exports[`UijettosOptionsButton component Renders correctly when supplied a css prefix 2`] = `
<div
  className="doodads-options-button"
  onBlur={[Function]}
  onFocus={[Function]}
>
  <button
    className="doodads-options-button__main-button"
  />
  <button
    className="doodads-options-button__options-trigger"
    onClick={[Function]}
  />
  <div
    className="doodads-options-button__options-list doodads-options-button__options-list--open"
  >
    <button
      className="doodads-options-button__option"
      onClick={[Function]}
    >
      widgets
    </button>
    <button
      className="doodads-options-button__option"
      onClick={[Function]}
    >
      doodads
    </button>
  </div>
</div>
`;

exports[`UijettosOptionsButton component Renders correctly when supplied with a label 1`] = `
<div
  className="uijettos-options-button"
  onBlur={[Function]}
  onFocus={[Function]}
>
  <button
    className="uijettos-options-button__main-button"
  >
    Foo Bar
  </button>
  <button
    className="uijettos-options-button__options-trigger"
    onClick={[Function]}
  />
</div>
`;

exports[`UijettosOptionsButton component Renders correctly when supplied with options 1`] = `
<div
  className="uijettos-options-button"
  onBlur={[Function]}
  onFocus={[Function]}
>
  <button
    className="uijettos-options-button__main-button"
  />
  <button
    className="uijettos-options-button__options-trigger"
    onClick={[Function]}
  />
</div>
`;

exports[`UijettosOptionsButton component Renders correctly when supplied with options 2`] = `
<div
  className="uijettos-options-button"
  onBlur={[Function]}
  onFocus={[Function]}
>
  <button
    className="uijettos-options-button__main-button"
  />
  <button
    className="uijettos-options-button__options-trigger"
    onClick={[Function]}
  />
  <div
    className="uijettos-options-button__options-list uijettos-options-button__options-list--open"
  >
    <button
      className="uijettos-options-button__option"
      onClick={[Function]}
    >
      widgets
    </button>
    <button
      className="uijettos-options-button__option"
      onClick={[Function]}
    >
      doodads
    </button>
  </div>
</div>
`;

exports[`UijettosOptionsButton component Renders options correctly when trigger button is clicked 1`] = `
<div
  className="uijettos-options-button"
  onBlur={[Function]}
  onFocus={[Function]}
>
  <button
    className="uijettos-options-button__main-button"
  />
  <button
    className="uijettos-options-button__options-trigger"
    onClick={[Function]}
  />
  <div
    className="uijettos-options-button__options-list uijettos-options-button__options-list--open"
  >
    <button
      className="uijettos-options-button__option"
      onClick={[Function]}
    >
      widgets
    </button>
    <button
      className="uijettos-options-button__option"
      onClick={[Function]}
    >
      doodads
    </button>
  </div>
</div>
`;
